xc/device.ml

```diff
@@ -2319,7 +2319,12 @@ module Backend = struct
     module Monitor = struct
       module Epoll = Core.Linux_ext.Epoll
       module Flags = Core.Linux_ext.Epoll.Flags
-      let create () = (Core.Or_error.ok_exn Epoll.create) ~num_file_descrs:1001 ~max_ready_events:1
+      let num_file_descrs =
+        match Core.Unix.RLimit.((get num_file_descriptors).cur) with
+        | Core.Unix.RLimit.Limit v -> Int64.to_int v
+        | Core.Unix.RLimit.Infinity -> Core.Int.max_value
```

Comment:
Would be better to raise an exception when the limit is infinity, otherwise it'd likely fail  quite soon when allocating memory for int_max file descriptors.

Reply:
Updated with raising an internal exn here.